internal/rbd/nodeserver.go

```diff
@@ -1190,26 +1197,54 @@ func (ns *NodeServer) xfsSupportsReflink() bool {
 func (ns *NodeServer) NodeGetVolumeStats(
 	ctx context.Context,
 	req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	var err error
-	targetPath := req.GetVolumePath()
-	if targetPath == "" {
-		err = fmt.Errorf("targetpath %v is empty", targetPath)
+	var (
+		imgInfo rbdImageMetadataStash
+		err     error
+	)
+	volumePath := req.GetStagingTargetPath()
+	if volumePath == "" {
+		if volumePath = req.GetVolumePath(); volumePath == "" {
+			return nil, status.Error(codes.InvalidArgument, "volume path must be provided")
+		}
+	} else {
+		volumePath = volumePath + "/" + req.GetVolumeId()
+		imgInfo, err = lookupRBDImageMetadataStash(volumePath)
+		if err != nil {
+			log.ErrorLog(ctx, "failed to find image metadata, at stagingPath: %s, err: %v", volumePath, err)
```

Comment:
use common formatting of errors, like `"failed to find image metadata at stagingPath %q: %v", volumePath, err`